packages/gatsby-source-drupal/src/utils.js

```diff
@@ -15,7 +95,9 @@ const handleReferences = (node, { getNode, createNodeId }) => {
       if (_.isArray(v.data)) {
         relationships[nodeFieldName] = _.compact(
           v.data.map(data => {
-            const referencedNodeId = createNodeId(data.id)
+            const referencedNodeId = createNodeId(
+              `${rootNodeLanguage}${data.id}`
```

Comment:
This seems to work for most cases. However when a entity doesn't support translation it seems that it's langcode is always the default language of the site. In my case this meant that file entities (which don't support translation) would be lose their relationships in languages other than the default. I was able to fix this in my fork of this patch but I will need to backport here.